zhmccli/_cmd_adapter.py

```diff
@@ -119,19 +147,26 @@ def adapter_show(cmd_ctx, cpc, adapter):
               help='The new MTU size of the adapter in KiB. '
               '(HiperSockets only).')
 @click.option('--allowed-capacity', type=int, required=False,
-              help='The maximum number of HBAs per partition. '
-              '(FCP only).')
+              help='The new maximum number of HBAs per partition. '
+              '(only certain adapter types).')
 @click.option('--chpid', type=str, required=False,
-              help='Channel path ID (CHPID, 2 hex chars) used by the '
-              'adapter\'s port. '
-              '(OSA, FICON, HiperSockets only).')
-@click.option('--crypto-number', type=int, required=False,
-              help='Identifier of the crypto adapter in the range 0-15 '
-              'Must be unique within the CPC. '
-              '(Crypto only).')
+              help='The new channel path ID (CHPID, 2 hex chars) used by the '
+              'adapter\'s partition. (OSA, FICON, HiperSockets only; Adapter '
+              'must not be used by any partition).')
 @click.option('--crypto-tke/--no-crypto-tke', default=None, required=False,
-              help='Permit TKE commands on the crypto adapter. '
-              '(Crypto only).')
+              help='The new setting whether TKE commands are permitted on the '
+              'crypto adapter. (Crypto only).')
+@click.option('--crypto-type', default=None, required=False,
+              type=click.Choice(CRYPTO_TYPE_OPTIONS),
+              help='The new crypto type (Crypto only; Adapter must be '
+              'offline).')
+@click.option('--ficon-type', default=None, required=False,
```

Comment:
I would prefer --storage-type or --storage-protocol (used in the WSAPI book). --ficon-type=fcp looks weird.

Reply:
The physical adapter card is named "FICON Express", and the input parameter of the "Change Adapter Type" operation is named "type". Hence I came up with the name "ficon-type". The corresponding property on the Storage Group object is also named "type".

I have changed it to "storage-type".